userspace/libpman/src/capture.c

```diff
@@ -178,6 +180,58 @@ int pman_get_scap_stats(void *scap_stats_struct)
 	return errno;
 }
 
+void* pman_get_scap_stats_v2(void *scap_stats_v2_struct, uint32_t flags, uint32_t* nstats, int32_t* rc)
+{
+	struct scap_stats_v2 *stats = (struct scap_stats_v2 *)scap_stats_v2_struct;
+	*nstats = 0;
+	if (!stats)
+	{
+		pman_print_error("pointer to scap_stats_v2 is empty");
+		*rc = errno;
+	}
+
+	if ((flags & PPM_SCAP_STATS_KERNEL_COUNTERS))
+	{
+		char error_message[MAX_ERROR_MESSAGE_LEN];
+		struct counter_map cnt_map;
+		int counter_maps_fd = bpf_map__fd(g_state.skel->maps.counter_maps);
+		if(counter_maps_fd <= 0)
+		{
+			*rc = errno;
+		}
+
+		/* We always take statistics from all the CPUs, even if some of them are not online.
+		* If the CPU is not online the counter map will be empty.
+		*/
+		for(uint32_t stat =  0;  stat < MODERN_BPF_MAX_KERNEL_COUNTERS_STATS; stat++)
+		{
+			stats[stat].type = STATS_VALUE_TYPE_U64;
+			stats[stat].flags = PPM_SCAP_STATS_KERNEL_COUNTERS;
+			stats[stat].value.u64 = 0;
+			strlcpy(stats[stat].name, modern_bpf_kernel_counters_stats_names[stat], STATS_NAME_MAX);
+		}
+
+		for(uint32_t index = 0; index < g_state.n_possible_cpus; index++)
+		{
+			if(bpf_map_lookup_elem(counter_maps_fd, &index, &cnt_map) < 0)
+			{
+				snprintf(error_message, MAX_ERROR_MESSAGE_LEN, "unbale to get the counter map for CPU %d", index);
+				pman_print_error((const char *)error_message);
+				close(counter_maps_fd);
+				*rc = errno;
```

Comment:
errno is invalid here, you need to:
* use `-ret`, where `ret` is the return value from `bpf_map_lookup_elem` and strerror() it into an error message
* not set *rc to an errno value anyway (it probably wants `SCAP_FAILURE`)

Also, you probably want an early return here.